tests/testthat/test-format_list_of.R

```diff
@@ -0,0 +1,4 @@
+test_that("smoke test", {
+  v <- vctrs::new_list_of(list(1L, 2:4), integer())
```

Comment:
I would add another test that also uses `new_list_of()` where one of the elements has a custom `size_sum()` method. That would enforce that you are indeed calling `size_sum()` in `pillar_shaft.vctrs_list_of()`, as opposed to what was being done previously.